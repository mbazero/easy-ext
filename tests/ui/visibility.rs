mod foo {
    use easy_ext::ext;

    #[ext(StrExt1)]
    impl str {
        fn method1(&self, pat: &str) -> String {
            self.replace(pat, "_")
        }
    }

    #[ext(StrExt2)]
    pub(self) impl str {
        fn method2(&self, pat: &str) -> String {
            self.replace(pat, "_")
        }
    }

    pub mod bar {
        use easy_ext::ext;

        #[ext(StrExt3)]
        pub(super) impl str {
            fn method3(&self, pat: &str) -> String {
                self.replace(pat, "_")
            }
        }
    }

    #[allow(unused_imports)]
    use bar::StrExt3;
}

fn main() {
    #[rustfmt::skip]
    use foo::StrExt1; //~ ERROR trait `StrExt1` is private [E0603]
    #[rustfmt::skip]
    use foo::StrExt2; //~ ERROR trait `StrExt2` is private [E0603]
    #[rustfmt::skip]
    use foo::bar::StrExt3; //~ ERROR trait `StrExt2` is private [E0603]
}
