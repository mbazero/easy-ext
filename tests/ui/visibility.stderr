error[E0603]: trait `StrExt1` is private
  --> tests/ui/visibility.rs:35:14
   |
35 |     use foo::StrExt1; //~ ERROR trait `StrExt1` is private [E0603]
   |              ^^^^^^^ private trait
   |
note: the trait `StrExt1` is defined here
  --> tests/ui/visibility.rs:5:5
   |
5  |     impl str {
   |     ^^^^^^^^

error[E0603]: trait `StrExt2` is private
  --> tests/ui/visibility.rs:37:14
   |
37 |     use foo::StrExt2; //~ ERROR trait `StrExt2` is private [E0603]
   |              ^^^^^^^ private trait
   |
note: the trait `StrExt2` is defined here
  --> tests/ui/visibility.rs:12:5
   |
12 |     pub(self) impl str {
   |     ^^^^^^^^^^^^^^^^^^

error[E0603]: trait `StrExt3` is private
  --> tests/ui/visibility.rs:39:19
   |
39 |     use foo::bar::StrExt3; //~ ERROR trait `StrExt2` is private [E0603]
   |                   ^^^^^^^ private trait
   |
note: the trait `StrExt3` is defined here
  --> tests/ui/visibility.rs:22:9
   |
22 |         pub(super) impl str {
   |         ^^^^^^^^^^^^^^^^^^^
